download/compiler_cache.go

```diff
@@ -34,7 +34,7 @@ func GetCompilerList(cacheDir string) (compilers types.Compilers, err error) {
 		fmt.Fprintln(os.Stderr, "updating compiler list...") // nolint:gas
 		err = UpdateCompilerList(cacheDir)
 		if err != nil {
-			return
+			fmt.Fprintln(os.Stderr, "failed to update compiler list")
```

Comment:
This should also print the `err` after the message and then set `err` to `nil` to clean up.

Reply:
You said err should be set to nil to clean up, but the variable `err` is reused again at the end of the file and replaced? So does this still need cleaning up?